src/EfficientDynamoDb/DynamoDbContext/IDynamoDbLowLevelContext.cs

```diff
@@ -35,7 +38,13 @@ public interface IDynamoDbLowLevelContext
         Task<DeleteItemResponse> DeleteItemAsync(DeleteItemRequest request, CancellationToken cancellationToken = default);
         
         Task<TransactWriteItemsResponse> TransactWriteItemsAsync(TransactWriteItemsRequest request, CancellationToken cancellationToken = default);
-        
+
+        Task<ExecuteStatementResponse> ExecuteStatementAsync(ExecuteStatementRequest request, CancellationToken cancellationToken = default);
+
+        Task<BatchExecuteStatementResponse> BatchExecuteStatementAsync(BatchExecuteStatementRequest request, CancellationToken cancellationToken = default);
+
+        Task<ExecuteTransactionResponse> ExecuteTransactionAsync(ExecuteTransactionRequest request, CancellationToken cancellationToken = default);
```

Comment:
I think that we need to introduce a thin wrapper over PartiQL methods to avoid low-level API pollution and provide better API discoverability. I'm thinking about something like:
```csharp
dbContext.LowLevel.PartiQL.ExecuteStatementAsync(...)
```
DynamoDB API PartiQL method names are too generic so I think having a proper name will make it easier to read the code that uses this API.

Reply:
What do you think about this?

```csharp
public interface IDynamoDbPartiQlContext
{
    Task<QueryResponse> QueryAsync(QueryRequest request, CancellationToken cancellationToken = default);

    Task<BatchItemsResponse> BatchItemsAsync(BatchItemsRequest request, CancellationToken cancellationToken = default);

    Task<TransactionItemsResponse> TransactionItemsAsync(TransactionItemsRequest request, CancellationToken cancellationToken = default);
}

internal class DynamoDbPartiQlContext : IDynamoDbPartiQlContext
{
    internal DynamoDbContextConfig Config { get; }
    internal HttpApi Api { get; }

    internal DynamoDbPartiQlContext(DynamoDbContextConfig config, HttpApi api)
    {
        Api = api;
        Config = config;
    }

    public Task<BatchItemsResponse> BatchItemsAsync(BatchItemsRequest request, CancellationToken cancellationToken = default)
    {
        // code
    }

    public Task<QueryResponse> QueryAsync(QueryRequest request, CancellationToken cancellationToken = default)
    {
        // code
    }

    public Task<TransactionItemsResponse> TransactionItemsAsync(TransactionItemsRequest request, CancellationToken cancellationToken = default)
    {
        // code
    }
}

public class DynamoDbLowLevelContext : IDynamoDbLowLevelContext
{
        // code
        private readonly DynamoDbPartiQlContext _partiQlContext;
        public IDynamoDbPartiQlContext PartiQL => _partiQlContext;

        internal DynamoDbLowLevelContext(DynamoDbContextConfig config, HttpApi api)
        {
        // code
            _partiQlContext = new DynamoDbPartiQlContext(config, api);
        }

        // code
}
```

Reply:
I'd rather leave original methods, requests and responses names (`ExecuteStatementAsync`, `ExecuteStatementRequest`,  `ExecuteStatementResponse`, etc) because:
- It matches names in DynamoDB docs so it might be easier for someone to associate official docs to library's implementation.
- PartiQL statement is not necessarily query - it can also be an update or delete expression. Also, I'd like to avoid potential confusion due to sharing the name with regular `QueryAsync()` request.

I also think that we should name the interface and implementation as `IDynamoDbLowLevelPartiQLContext`:
- It should have `LowLevel` in the name because most likely there will be a high level version of PartiQL API later.
- Lowercase `l` in `Ql` looks weird and can be confused as uppercase `I`. So I think in this case it's better to use full acronym in the name as `PartiQL`.

Other than naming stuff, it looks great.